clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -561,8 +561,8 @@ static void collectSYCLAttributes(Sema &S, FunctionDecl *FD,
     llvm::copy_if(FD->getAttrs(), std::back_inserter(Attrs), [](Attr *A) {
       // FIXME: Make this list self-adapt as new SYCL attributes are added.
       return isa<IntelReqdSubGroupSizeAttr, IntelNamedSubGroupSizeAttr,
-                 ReqdWorkGroupSizeAttr, SYCLIntelKernelArgsRestrictAttr,
-                 SYCLIntelNumSimdWorkItemsAttr,
+                 ReqdWorkGroupSizeAttr, WorkGroupSizeHintAttr,
```

Comment:
https://github.com/intel/llvm/blob/5410e2381da3caac358b4cac3a673bb102657684/clang/lib/Sema/SemaSYCL.cpp#L3971

@Michoumichmich, you need to add case attr::Kind::WorkGroupSizeHintAttr: